hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/RestCallCollector.java

```diff
@@ -0,0 +1,191 @@
+        }
+
+        @Override
+        public int compareTo(RestCallCollector.RequestIdentifier o) {
+            return COMPARATOR.compare(this, o);
+        }
+
+        public String getMethod() {
+            return method;
+        }
+
+        public String getPath() {
+            return path;
+        }
+    }
+
+    private final MwCounter mapPostSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapPostFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapGetSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapGetFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapDeleteSuccCount = MwCounter.newMwCounter();
+    private final MwCounter mapDeleteFailCount = MwCounter.newMwCounter();
+    private final MwCounter mapTotalRequestCount = MwCounter.newMwCounter();
+    private final MwCounter queuePostSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queuePostFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueGetSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queueGetFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueDeleteSuccCount = MwCounter.newMwCounter();
+    private final MwCounter queueDeleteFailCount = MwCounter.newMwCounter();
+    private final MwCounter queueTotalRequestCount = MwCounter.newMwCounter();
+
+    private final MwCounter requestCount = MwCounter.newMwCounter();
+    private final ConcurrentSkipListSet<RequestIdentifier> uniqueRequests = new ConcurrentSkipListSet<>();
+    private final ConcurrentSkipListSet<String> accessedMaps = new ConcurrentSkipListSet<>();
+    private final ConcurrentSkipListSet<String> accessedQueues = new ConcurrentSkipListSet<>();
+
+    void collectExecution(RestCallExecution execution) {
+        updateRequestCounters(execution);
+        boolean isMap = "map".equals(execution.getObjectType());
+        boolean isQueue = "queue".equals(execution.getObjectType());
+        String objectName = execution.getObjectName();
+        updateAccessedObjectSets(isMap, isQueue, objectName);
+        if (execution.getMethod().equalsIgnoreCase("post")) {
+            handlePost(execution.isSuccess(), isMap, isQueue);
+        } else if (execution.getMethod().equalsIgnoreCase("get")) {
+            if (isMap) {
+                (execution.isSuccess() ? mapGetSuccCount : mapGetFailCount).inc();
+            } else if (isQueue) {
+                (execution.isSuccess() ? queueGetSuccCount : queueGetFailCount).inc();
+            }
+        } else if (execution.getMethod().equalsIgnoreCase("delete")) {
+            if (isMap) {
+                (execution.isSuccess() ? mapDeleteSuccCount : mapDeleteFailCount).inc();
+            } else if (isQueue) {
+                (execution.isSuccess() ? queueDeleteSuccCount : queueDeleteFailCount).inc();
+            }
+        }
+    }
+
+    private void handlePost(boolean isSuccess, boolean isMap, boolean isQueue) {
```

Comment:
why is post in a separate method? It seems to be the same structurally as get and delete